public/js/components/SourceSearch.js

```diff
@@ -61,27 +57,27 @@ const Search = React.createClass({
 
   toggle(key, e) {
     e.preventDefault();
-    this.setState({ searchOn: !this.state.searchOn });
+    this.props.toggleSearch(!this.props.searchOn);
   },
 
   onEscape(shortcut, e) {
-    if (this.state.searchOn) {
+    if (this.props.searchOn) {
       e.preventDefault();
-      this.setState({ searchOn: false });
+      this.props.toggleSearch(false);
     }
   },
 
   close() {
-    this.setState({ searchOn: false });
+    this.props.toggleSearch(false);
   },
 
   render() {
-    return this.state.searchOn ?
+    return this.props.searchOn ?
       dom.div({ className: "search-container" },
       Autocomplete({
         selectItem: result => {
           this.props.selectSource(result.id);
-          this.setState({ searchOn: false });
+          this.props.toggleSearch(false);
```

Comment:
Hmm, I'm not thrilled about refactoring SourcesState so that the searchOn state is kept in the app. I can see the argument for it being app state, but at this point we have a nice convention for state being local to the component it is being used.

I think it would be nice to use DOM events to open the panel. This way, the panel API is largely DOM and doesn't rely on React.

Here is a function for opening the panel that could be use in the tab onClick.
```js
function openSearchPanel() {
  window.dispatchEvent(Object.assign(new Event("keydown"), {
    key: "p", metaKey: true, ctrlKey: false, altKey: false, shiftKey: false
  }))
}
```